coderd/prometheusmetrics/insights/metricscollector.go

```diff
@@ -143,18 +163,62 @@ func (mc *MetricsCollector) Collect(metricsCh chan<- prometheus.Metric) {
 		return // insights data not loaded yet
 	}
 
+	// Custom apps
+	for _, appRow := range data.apps {
+		displayName := appRow.DisplayName.String
+		if displayName == "" { // just in case Prometheus complains about missing label value
+			displayName = "unknown"
+		}
+		metricsCh <- prometheus.MustNewConstMetric(applicationsUsageSecondsDesc, prometheus.GaugeValue, float64(appRow.UsageSeconds), data.templateNames[appRow.TemplateID], displayName)
+	}
+
+	// Built-in apps
+	for _, templateRow := range data.templates {
+		if templateRow.UsageVscodeSeconds > 0 {
```

Comment:
Why do we skip the rows if usage is 0? Isn't it valid to push zero values too? For instance, what if we go from 100 to 0, will this ever reset?

Reply:
This is a collector, so if we create const metrics every time, they won't be present in the Prometheus response unless created.

Anyway, I was on the fence about whether or not publish zero metrics here, so I will remove conditions.

Reply:
Hmm, won’t the absence show up as null then instead of zero in graphs? Semantically I think it’s valid to include zero values.